app/models/risk_condition.rb

```diff
@@ -1,35 +1,50 @@
 class RiskCondition < ApplicationRecord
   serialize :parameters, Hash
+  enum condition_type: [:grace_day_usage, :grade_drop, :no_submissions, :low_grades]
 
   has_many :watchlist_instances, dependent: :destroy
   belongs_to :course
 
-  GRACE_DAY_USAGE = 1 # :grace_day_threshold, :date
-  GRADE_DROP = 2 # :percentage_drop, :consecutive_counts
-  NO_SUBMISSIONS = 3 # :no_submissions_threshold
-  LOW_GRADES = 4 # :grade_threshold, :count_threshold
+  # parameter correspondence:
+  # :grace_day_usage => :grace_day_threshold, :date
+  # :grade_drop => :percentage_drop, :consecutive_counts
+  # :no_submissions => :no_submissions_threshold
+  # :low_grades => :grade_threshold, :count_threshold
 
   def self.create_condition_for_course_with_type(course_id, type, params)
-  	# parameter check shouldn't surface to user and is for debug only
-    if (type == GRACE_DAY_USAGE && (params[:grace_day_threshold].nil? || params[:date].nil? || params.length != 2)) ||
-       (type == GRADE_DROP && (params[:percentage_drop].nil? || params[:consecutive_counts].nil? || params.length != 2)) ||
-       (type == NO_SUBMISSIONS && (params[:no_submissions_threshold].nil? || params.length != 1)) ||
-       (type == LOW_GRADES && (params[:grade_threshold].nil? || params[:count_threshold].nil? || params.length != 2))
+  	# parameter check shouldn't surface to user and is for sanity check during development only
```

Comment:
I was adding new condition using this function, but I realize there might be some issue here?
Because when the user updates the condition, they update all four conditions in one go, not one by one. And they might deselect (remove) some conditions as well.

For example:
**First time saving**: The user sets grace_days_usage, and low grades.
**Second time saving**: The user sets no_submissions, and low grades, and unselects grace_days_usage.
So in this case I forsee grace_days_usage being stuck at v1? 

Since I think this is to be addressed in your separate ticket, and I will proceed to approve the PR. That way you can address this in your next PR instead. (Maybe it is how you intend this to be? I'm not sure)